hw/top_earlgrey/data/clocks.xdc

```diff
@@ -30,7 +30,15 @@ set_clock_sense -positive top_*/u_clkmgr_aon/u_no_scan_io_div2_div/gen_div2.u_di
 ## JTAG and SPI clocks
 create_clock -add -name lc_jtag_tck -period 100.00 -waveform {0 5} [get_pin top_*/u_pinmux_aon/u_pinmux_strap_sampling/u_pinmux_jtag_buf_lc/prim_clock_buf_tck/gen_xilinx.u_impl_xilinx/gen_fpga_buf.bufg_i/O]
 create_clock -add -name rv_jtag_tck -period 100.00 -waveform {0 5} [get_pin top_*/u_pinmux_aon/u_pinmux_strap_sampling/u_pinmux_jtag_buf_rv/prim_clock_buf_tck/gen_xilinx.u_impl_xilinx/gen_fpga_buf.bufg_i/O]
-create_clock -add -name clk_spi_in  -period 100.00 -waveform {0 5} [get_pin top_*/u_spi_device/u_clk_spi_in_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.bufg_i/O]
-create_clock -add -name clk_spi_out -period 100.00 -waveform {0 5} [get_pin top_*/u_spi_device/u_clk_spi_out_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.bufg_i/O]
+create_clock -add -name clk_spi  -period 100.00 -waveform {0 50} [get_ports SPI_DEV_CLK]
```

Comment:
@a-will , this seemed to really blow up CI time.. not sure if I just got unlucky.
I don't see anything really out of wack here though.

Reply:
For anyone else who reads it: This came down to a clocking architecture that built in difficult-to-solve hold time problems. clk_spi_out and clk_spi_in had the same ultimate source, but clk_spi_out's path included another clock buffer. Consequently, signals traveling from clk_spi_in to clk_spi_out could arrive "from the future", as clk_spi_out was delayed enough (multiple nanoseconds) to clock in data coming from the same clk_spi_in cycle.